common/djangoapps/django_comment_common/models.py

```diff
@@ -105,3 +109,32 @@ class Meta:
 
     def __unicode__(self):
         return self.name
+
+def all_permissions_for_user_in_course(user, course_id):
+    course_blacked_out_key = "django_comment_common.models.course_blacked_out.{}".format(course_id)
+    course_blacked_out = cache.get(course_blacked_out_key)
+    if course_blacked_out is None:
+        course = modulestore().get_course(course_id)
```

Comment:
If, rather than using `modulestore().get_course(course_id)`, you used `runtime.get_block(...)` with the correct usage id for the course, then multiple calls in the same request (at least in split) would return the same object. In fact, I would have expected split to do the appropriate caching so that calling `get_course` multiple times in the same request wouldn't be doing multiple mongo hits.